### PR TITLE
Compat SOG - Add barrel swap for MGs and belt linking for drum magazines

### PR DESCRIPTION
**Pull request by Esmeray6**
**When merged this pull request will:**
- Add barrel swapping to the following SOGPF weapons: MG42 and L4.
- Change the drum magazines for MG42 to be considered belts. This will allow belt linking with these magazines.

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/compat_sog/CfgMagazines/belts.hpp [hunk 1]
```diff
@@ -12,3 +12,8 @@ class vn_m16_mag_base;
 class vn_m63a_100_mag: vn_m16_mag_base {
     ACE_isBelt = 1;
 };
+class vn_mg42_50_mag: vn_lmgmag_base {
+    ACE_isBelt = 1;
+};
+class vn_mg42_50_t_mag: vn_mg42_50_mag {};
+class vn_rpd_125_mag: vn_rpd_100_mag {};
```
Comment (PabstMirror) [inline addons/compat_sog/CfgMagazines/belts.hpp:19]:
```suggestion
```
ok to drop these imho

Comment (Esmeray6) [reply addons/compat_sog/CfgMagazines/belts.hpp:19]: True. Not sure what I was thinking when I added them. Purposes of explicitness, I suppose.

**Comment by Drofseh** [comment]
RPD does not have a removable barrel.

**Comment by Drofseh** [comment]
Neither does the M1918 BAR

**Comment by Esmeray6** [comment]
> RPD does not have a removable barrel.
> Neither does the M1918 BAR

Ah, right. I was thinking more in terms of player convenience at the time. I'll remove them.

File: addons/compat_sog/CfgMagazines/belts.hpp [hunk 2]
```diff
@@ -12,3 +12,6 @@ class vn_m16_mag_base;
 class vn_m63a_100_mag: vn_m16_mag_base {
     ACE_isBelt = 1;
 };
+class vn_mg42_50_mag: vn_lmgmag_base {
```
Comment (LinkIsGrim) [inline addons/compat_sog/CfgMagazines/belts.hpp:15]: I'm not a fan of this. Can MG42 drums be linked?

Comment (Esmeray6) [reply addons/compat_sog/CfgMagazines/belts.hpp:15]: If I understand it correctly, [Wikipedia](https://en.wikipedia.org/wiki/MG_42#:~:text=Feeding,-Method%20of%20joining&text=The%20MG%2042%20could%20use,housed%20in%20the%20feed%20cover.) says they cannot. Only through additional connectors, possibly.

Of course, it's not a standard practice in that case. Most of the time they would be loaded with 50-round drum magazines on the go.

Comment (Drofseh) [reply addons/compat_sog/CfgMagazines/belts.hpp:15]: They can, but only in 50 round segments (ei. a 200 round belt would have 4 segments).

I would keep them as linkable, it's not significantly less realistic than how linking any other belt can't exceed the `count` of the magazine class.

https://cdn11.bigcommerce.com/s-817fb/images/stencil/500x659/products/5331/69865/IMG_5654__11640.1694970702.jpg

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**